modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -555,16 +555,16 @@ def join(self, other, how="inner", on=None, sort=False, suffixes=("_x", "_y")):
 
         conflicting_cols = set(self.columns) & set(other.columns) - set(on)
         for c in self.columns:
-            suffix = suffixes[0] if c in conflicting_cols else ""
-            new_columns.append(c + suffix)
+            new_name = f"{c}{suffixes[0]}" if c in conflicting_cols else c
```

Comment:
@ienkovich , based on that comment ([link](https://github.com/modin-project/modin/pull/3149#discussion_r659730317)), do we need to convert `c` to `str` if `not isinstance(c, str)`?

Reply:
We do conversion here to apply suffixes. We don't have to do it if suffixes are not applied (OmniSci will work with mangled names).

Reply:
I mean OmniSci is able to work with non-string columns?

Reply:
We mangle column names before passing data and queries to OmniSci.

Reply:
I see, thanks!